activerecord/test/cases/relation/or_test.rb

```diff
@@ -138,5 +138,16 @@ def test_or_with_scope_on_association
         author.top_posts.or(author.other_top_posts)
       end
     end
+
+    def test_structurally_incompatible_values
+      assert_nothing_raised do
+        Post.includes(:author).includes(:author).or(Post.includes(:author))
+        Post.eager_load(:author).eager_load(:author).or(Post.eager_load(:author))
+        Post.preload(:author).preload(:author).or(Post.preload(:author))
+        Post.group(:author).group(:author).or(Post.group(:author))
```

Comment:
`group` doesn't support association name, this will generate invalid (unknown column) SQL.

```suggestion
        Post.group(:author_id).group(:author_id).or(Post.group(:author_id))
```

Reply:
Oops. I fixed it.